### PR TITLE
Allow unicode in title

### PR DESCRIPTION
**Pull request by yipenghe**
#1436 See latest comment from me below
<s>Supporting unicode in title & name for new frontend, the trick is to encode the way special characters get encoded in url and store that to database. 

The root cause of unicode error seems to be related to database, I tried several ways to change the character set of the database, but character-set-server is still latin1. It seems to be tricky when we are trying to change an existing database's character set...</s>
Screenshot from 2019-10-16 11-24-41

### CODE REVIEW

File: codalab/objects/worksheet.py
Comment (percyliang) [inline codalab/objects/worksheet.py]: Why are these removed?

File: docker/compose_files/files/mysqlconf/mysql.cnf
Comment (percyliang) [inline docker/compose_files/files/mysqlconf/mysql.cnf]: Do we need this to work?  We have to be careful, because on prod, the mysql server is setup not through docker but is on another server.

File: frontend/src/components/EditableField.js
Comment (percyliang) [inline frontend/src/components/EditableField.js]: These are not necessarily URIs though, right?

Comment (yipenghe) [reply frontend/src/components/EditableField.js]: Yeh, should actually be using encodeURIComponent, which is safe to use on strings

File: docker/compose_files/files/mysqlconf/mysql.cnf
Comment (yipenghe) [reply docker/compose_files/files/mysqlconf/mysql.cnf]: Added accidentally, will remove

File: codalab/objects/worksheet.py
Comment (yipenghe) [reply codalab/objects/worksheet.py]: These check if the name contains ASCII patterns

Comment (percyliang) [reply codalab/objects/worksheet.py]: I think the worksheet name should still be restricted (like an identifier).  The title can be free-form.

**Review by percyliang — COMMENTED** [review]
Just to clarify: only the title should be unicode / arbitrary.  The name (for both bundles and worksheets) should still be constrained.

**Comment by yipenghe** [comment]
> Just to clarify: only the title should be unicode / arbitrary. The name (for both bundles and worksheets) should still be constrained.

Ok, I will update accordingly

**Review by epicfaace — CHANGES_REQUESTED** [review]
I would suggest using the `encode_str` and `decode_str` functions for consistency with how we handle storing worksheet item contents with unicode in them. (Otherwise, we will have two workarounds in place -- for one, we will be using `encode_str` and `decode_str` to store worksheet item content, and two, we will store uri-encoded versions of unicode text for worksheet titles, which is the approach in this PR. It seems better to pick one way of doing this and be consistent, so that in the future, we can more easily change the unicode encoding method altogether if need be.)

**Comment by yipenghe** [comment]
Well I'm definitely much more familiar with all these codes now...
1. removed all the uri-escaping encode
2. use bundle_model level encode&decode str for encoding unicode, this allows both web interface editting & cli editting (using the wedit command)
3. still restrict worksheet name to be ascii characters only & follow name pattern
Screenshot from 2019-10-22 01-44-07
Screenshot from 2019-10-22 01-43-38

File: codalab/model/bundle_model.py
Comment (yipenghe) [inline codalab/model/bundle_model.py]: This takes care of general worksheet fetching

File: codalab/model/bundle_model.py
Comment (yipenghe) [inline codalab/model/bundle_model.py:1273]: This takes care of worksheet fetching through 'wsearch' command line, the None check is needed as some legacy title might be None for some reason (i.e before didn't prevent title being None)

File: frontend/src/components/EditableField.js
Comment (yipenghe) [inline frontend/src/components/EditableField.js:88]: This change makes EditableField togglable for Ascii restriction

File: frontend/src/components/EditableField.js
Comment (percyliang) [inline frontend/src/components/EditableField.js]: nit: add space after `//`

File: frontend/src/components/EditableField.js
Comment (percyliang) [inline frontend/src/components/EditableField.js]: nit: put space before '?' and ':'

**Review by percyliang — APPROVED** [review]
Could you add a comment in front of `decode_str` and `encode_str` saying how these functions are meant to be used (needed right now because we want to use unicode but DB stores serialized bytes...).  Currently, these are used only for worksheet title and worksheet lines, right?  I feel like inserting these calls might be a bit error-prone since it's very decentralized.

**Comment by yipenghe** [comment]
> Could you add a comment in front of `decode_str` and `encode_str` saying how these functions are meant to be used (needed right now because we want to use unicode but DB stores serialized bytes...). Currently, these are used only for worksheet title and worksheet lines, right? I feel like inserting these calls might be a bit error-prone since it's very decentralized.

Updated. It seems like the unicode change is failing a test for unicode constraints? Where should I change that test/ can I still merge directly?

**Comment by percyliang** [comment]
`test_cli.py`?  Always fix tests before merging.

**Comment by yipenghe** [comment]
@percyliang Actually, do we want this fix for master as well or is it good to have it released along with new frontend? I can open a new PR for fixing it in master since this is working

**Comment by percyliang** [comment]
Ah, would have been good to merge into master, if it's not too much work.

File: test_cli.py
Comment (epicfaace) [inline test_cli.py:1680]: @yipenghe You didn't uncomment out the second part of the test, which actually checks that the title is shown as unicode. Can you uncomment this line and make a PR?

Comment (yipenghe) [reply test_cli.py:1680]: Good catch, will do! Thanks!

**Pull request merged**
